### ISSUE TITLE
Potential false positives for DOC202

### ISSUE THREAD

**Issue by m00rex** [astral-sh/ruff#25412]
### Question

When a function has an early bare `return` and a `Returns` section documenting that early return, Ruff reports `DOC202`:

```python
def validate_name(name: str | None) -> None:
    """A method to validate a name.

    Args:
        name (str | None): The name to validate or None.

    Returns:
        None: Early return if the name is empty or None.
    """
    if not name:
        return

    print(name)
```

```bash
ruff check --output-format concise --preview --select DOC202 test.py
test.py:2:5: error[DOC202] Docstring should not have a returns section because the function doesn't return anything
Found 1 error.
```

However, `pydoclint` does not report a violation for the same file:

```bash
pydoclint test.py
Skipping files that match this pattern: \.git|\.tox
test.py
🎉 No violations 🎉
```

If I change the early return from `return` to `return None`, Ruff no longer reports `DOC202`.
Instead Ruff reports `RET501`, because explicitly returning `None` is discouraged when `None` is the only possible return value.

This creates an awkward interaction between the two rules as `DOC202` is avoided by writing `return None` and `RET501` is avoided by writing `return`.

I would expect Ruff to either match `pydoclint`'s behavior here, or clarify whether `DOC202` intentionally considers a bare early `return` in a `-> None` function as incompatible with a `Returns` section.

### Version

0.15.14

**Comment by ntBre** [comment]
Thanks for the report! 

I'm honestly not that familiar with all the Python docstring conventions, but I kind of feel like DOC202 is reasonable here, personally. Is it common to document early returns in a `Returns` section, even if the function always returns `None`?

I found https://github.com/astral-sh/ruff/issues/21336, which I initially thought was closely related, but at least in this exact example, I feel like better fix is probably removing the `Returns` section rather than modifying the `return` statement itself. That should make both rules happy.

We could probably clarify this in the [documentation](https://docs.astral.sh/ruff/rules/docstring-extraneous-returns/#why-is-this-bad), which I think is using "explicit" as the opposite of the internal `ImplicitNone`, which includes bare `return`:

https://github.com/astral-sh/ruff/blob/3cb09eba689ebb49e799131092121928cc789c18/crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs#L839-L843

but a bare `return` still seems "explicit" in the normal usage of the word.

**Comment by m00rex** [comment]
It's just a style we agreed to internally. We can propably change it, as we are still rolling out ruff for all repositories. It's also just a preview rule which we do not check for but I check sometimes to find new ruff suggestions.

I think a bare `return` is as explizit as a `return None`. In the case of `return None` we always document `| None` which I find is pretty similar.

But the current state is a conflict of documentation and two rules. Even when only updating the documentation these two rules would still suggest different things. Which could trigger `RET501` on the first and `DOC202` on the second ruff check.

**Comment by m00rex** [comment]
First ruff check:
```bash
ruff check --output-format concise --preview --select RET501,DOC202 test.py      
test.py:11:9: error[RET501] [*] Do not explicitly `return None` in function if it is the only possible return value
Found 1 error.
[*] 1 fixable with the `--fix` option.
```
Second ruff check:
```bash
ruff check --output-format concise --preview --select RET501,DOC202 test.py --fix
test.py:2:5: error[DOC202] Docstring should not have a returns section because the function doesn't return anything
Found 2 errors (1 fixed, 1 remaining).
```

**Comment by ntBre** [comment]
Thanks, yes I understand the situation. I should have included the playground link I was trying this out in: https://play.ruff.rs/42e1e816-24ca-43bd-a3d7-c81e3f4fe343. I don't think the rules are necessarily in conflict because the third example in the playground link satisfies both rules by deleting the `Returns` section, which I think is fair for `DOC202` to call extraneous. But I'm open to other opinions.

The diagnostic for `DOC202` also says:

> Docstring should not have a returns section because the function doesn't return anything

so it's not suggesting that you add `return None` and violate `RET501`. I think I agree with you that `return` and `return None` should be treated the same way here, though, as also mentioned in https://github.com/astral-sh/ruff/issues/21336.
